src/applications/disability-benefits/526EZ/components/AsyncDisplayWidget.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+
+import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
+
+const PENDING = 'PENDING';
+const RESOLVED = 'RESOLVED';
+const REJECTED = 'REJECTED';
+
+
+/**
+ * Handles the various display statuses when calling an asynchronous function.
+ */
+export default class AsyncDisplayWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      promiseState: PENDING
+    };
+
+    // Make sure we configured it correctly
+    // StringField passes 'ui:options' as the options prop
+    const uiOptions = props.options;
+    if (!uiOptions) {
+      throw new Error('No ui:options supplied to AsyncDisplayWidget.');
+    }
+
+    if (typeof uiOptions.viewComponent !== 'function') {
+      throw new Error('AsyncDisplayWidget requires viewComponent in ui:options to be a React component.');
+    }
+
+    if (typeof uiOptions.callback !== 'function') {
+      throw new Error('AsyncDisplayWidget requires callback in ui:options to be a function.');
+    }
+
+    // TODO: Don't call the callback _every_ time the component is mounted
```

Comment:
Why not? Should we use redux to store the state to decide whether it is necessary?

Reply:
Yeah, I oscillated back and forth on this. I think we can do a sort of cache of the result because if they go to the next page, accidentally hit the back button, and then go forward again, we probably shouldn't query the api. I think the right answer is using redux to store whether the query has happened or not, but that gets complicated because we'd then have to store the state, etc. I decided to not go down that route for now in favor of optimizing it later if we see a problem.